### PR TITLE
Fix issue where preview replicaset was not scaled properly during a scaling event

### PR DESCRIPTION
**Pull request by jessesuen**
Resolves https://github.com/argoproj/argo-rollouts/issues/81

Working on unit test for this

### CODE REVIEW

File: controller/bluegreen.go [hunk 1]
```diff
@@ -248,6 +248,20 @@ func (c *Controller) scaleBlueGreen(rollout *v1alpha1.Rollout, newRS *appsv1.Rep
 		}
 	}
 
+	previewRS, _ := replicasetutil.GetReplicaSetByTemplateHash(allRS, rollout.Status.BlueGreen.PreviewSelector)
+	if previewRS != nil {
+		var previewReplicas int32
+		if rollout.Spec.Strategy.BlueGreenStrategy.PreviewReplicaCount != nil {
```
Comment (dthomson25) [inline controller/bluegreen.go]: We need to take into account when the `status.blueGreenStatus.scaleUpPreviewCheckPoint` field is set to true. That field is used to indicates that the rollout has been unpaused and it should scale up the preview ReplicaSet to the full Replica count before switching the active service to it.

File: controller/bluegreen.go [hunk 2]
```diff
@@ -248,6 +248,20 @@ func (c *Controller) scaleBlueGreen(rollout *v1alpha1.Rollout, newRS *appsv1.Rep
...
+			previewReplicas = *rollout.Spec.Strategy.BlueGreenStrategy.PreviewReplicaCount
+		} else {
+			previewReplicas = rolloutReplicas
```
Comment (dthomson25) [inline controller/bluegreen.go]: Is there a reason that you aren't setting the previewReplicas to rolloutReplicas as the default and simplifying the if else statement to just an if statement?

File: controller/bluegreen.go (hunk 1 above)
Comment (jessesuen) [reply controller/bluegreen.go]: Thanks I thought about that but didn't know if there was a good way to know.

File: controller/bluegreen.go (hunk 2 above)
Comment (jessesuen) [reply controller/bluegreen.go]: Good suggestion.

**Pull request merged**